oauth2-server/src/main/java/com/clouway/oauth2/ClientAuthorizationActivity.java

```diff
@@ -49,7 +49,7 @@ public Response execute(String identityId, Request request) {
     }
 
     Set<String> scopes = Sets.newTreeSet(Splitter.on(" ").omitEmptyStrings().split(scope));
-    Optional<Authorization> possibleAuthorizationResponse = clientAuthorizationRepository.authorize(client, identityId, scopes, responseType);
+    Optional<Authorization> possibleAuthorizationResponse = clientAuthorizationRepository.authorize(client, identity.idenityId, identity.scopes, responseType);
```

Comment:
we have the flexibility to choose which scopes to return
we have the request(requested scopes in it) and we have the SID and client id - so we can find the exact user with the exact scopes for the application(clientId) and we can do what ever we wont with them in the omplementation in the accounts app.